### ISSUE TITLE
XCS trial run suggestions / TODO

### ISSUE THREAD

**Issue by klauer** [pcdshub/las-dispersion-scan#6]
- [x] Export button to save plots
- [x] Number of spectra per point widget + scan modification (averaged)
- [x] Wait on Aaron to decide which parameters can/should be shown; hide the rest or put them in an advanced panel
- [ ] Descriptions of retrieval parameters (like what is "gratinga") - tooltip additions (Aaron?)

**Comment by slactjohnson** [comment]
Additional feedback/info from testing 11/3: 

- Averaging N spectra at each motor position is extremely desirable. 
- The position of the optimum pulse length is currently returned on the plot in meters, and is rounded off. We should change this to mm. Here is the line where this happens: https://github.com/pcdshub/las-dispersion-scan/blob/master/las_dispersion_scan/plotting.py#L184
- Aaron is working with another laser scientist to develop their own retrieval algorithm so that we can move away from PyPret and make this a fully open project. 
- This is an expert level tool, so they would like to keep all the customization options that we have available on the screen. We could perhaps at a later date develop a simpler "user" version of the screen. Aaron was interested in adding Tooltip information, so we could work with him to make these more informative.
- Default Plot to acquired, not retrieved
- Fix percent complete; progress bar goes to 100% and then hangs for a while
- Scan padding does weird things when the wavelength range is smaller than the padding. This should be set to zero. https://github.com/pcdshub/las-dispersion-scan/blob/0049619f49e7f0b5f84c6aaa9dd7fd7a252a49a7/las_dispersion_scan/plotting.py#L300
- Add ability to plot the reconstructed pulse at any motor position in the scan. Sometimes they want to know the pulse length at some other off-optimum position.

**Comment by klauer** [comment]
>  Fix percent complete; progress bar goes to 100% and then hangs for a while

This hangs because matplotlib plots are rather slow. I suppose it could be hidden prior to the updates, but the UI will still lag.

>  Scan padding does weird things when the wavelength range is smaller than the padding. This should be set to zero. 

Should this be configurable or always just 0?

**Comment by slactjohnson** [comment]
I figured as much, not a big deal; the progress bar can be ignored. 

I think this can be zero always.

**Comment by klauer** [comment]
>  Add ability to plot the reconstructed pulse at any motor position in the scan. Sometimes they want to know the pulse length at some other off-optimum position.

I thought it would be obvious in the code which array to plot, but it wasn't. Perhaps Aaron can provide a sample and I'll just port it in here.

**Comment by klauer** [comment]
Almost all of these are done. I'll open a new issue just about the remaining tooltip thing
